docs/core/deploying/ready-to-run.md

```diff
@@ -0,0 +1,65 @@
+
+To compile your project as ReadyToRun, do the following:
+
+01. Add the `<PublishReadyToRun>` setting to your project:
+
+    ```xml
+    <PropertyGroup>
+      <PublishReadyToRun>true</PublishReadyToRun>
+    </PropertyGroup>
+    ```
+
+02. Publish an application. For example, this command creates an app for the 64-bit version of Windows:
+
+    ```dotnetcli
+    dotnet publish -c Release -r win-x64
+
+    ```
+
+## Impact of using the ReadyToRun feature
+
+### General impact
+Ahead of time compilation has somewhat complex and unpredictable performance impacts on application performance. In general the size of an assembly will grow to between 2 to 3 times larger. This increase in the physical size of the file may reduce the performance of loading the assembly from disk, and increase working set of the process. However, in return the number of methods compiled is typically reduced substantially. The result is that most applications that consist of a substantial amount of code receive large performance benefits from enabling ReadyToRun. Applications which have small amounts of code will likely not experience a significant improvement from enabling ReadyToRun, as the .NET framework assemblies have already been precompiled with ReadyToRun.
+
+Note that the startup improvement discussed here applies not only to application startup, but also to the first use of any code in the application. For instance, ReadyToRun can be used to reduce the response latency of the first use  of a WebAPI in an ASP.NET application.
+
+### Interaction with tiered compilation
+Ahead of time code generation quality is generally slightly worse than the quality of code produced by the JIT. To address this issue, tiered compilation will replace commonly used ReadyToRun methods with JIT generated methods.
+
+## How is the set of precompiled assemblies chosen?
+The SDK will precompile the assemblies that are distributed with the application. For self-contained applications this will include recompiling the framework. C++/CLI binaries are not eligible for ReadyToRun compilation.
+
+## How is the set of methods to precompile chosen?
+The compiler will attempt to pre-compile as many methods as it can. However various reasons it is not expected that using the ReadyToRun feature will result in preventing the JIT from executing.
+
+Such reasons may include, but are not limited to:
+
+- Use of generic types defined in separate assemblies
+- Interop with native code
+- Use of hardware intrinsics which the compiler cannot prove are safe to use at publish time
```

Comment:
> intrinsics which the compiler cannot prove are safe

Nit: I think technically we should use 'that' instead of 'which', because some intrinsics are considered safe.

Reply:
One more nit, it may be misread as 'cannot prove are (safe to use at publish time)' instead of 'cannot prove are (safe to use) at publish time'. Maybe we should reword it, e.g.,
> cannot prove are safe to use on a target machine